cuqi/distribution/_distribution.py

```diff
@@ -267,11 +266,32 @@ def _condition(self, *args, **kwargs):
                     # Define new partial function with partially defined args
                     func = partial(var_val, **var_args)
                     setattr(new_dist, var_key, func)
-
-        # If conditioning on the name of the distribution
-        # we convert the distribution to a likelihood.
-        if self.name in kwargs:
-            return new_dist.to_likelihood(kwargs[self.name])               
+                
+                # Store processed keywords
+                processed_kwargs.update(var_args.keys())
+
+        # Check if more arguments were passed than conditioning variables
+        # If so we assume the last one is the distribution name
```

Comment:
This looks like a reasonable convention. I wanted to ask if it is it documented for the user that the last argument is the calling distribution random variable realization (name).

Reply:
It is now :)